spec/requests/api/provision_requests_spec.rb

```diff
@@ -105,6 +105,140 @@ def app
     end
   end
 
+  context "AWS advanced provision requests" do
+    let(:aws_dialog)     { FactoryGirl.create(:miq_dialog_aws_provision) }
```

Comment:
You are creating the dialog here but the provisioning workflow uses automate to define the dialog name to use.  This allows a user to update the automate modeling to select different dialogs as need.  For example based on the tags applied to the user.

Along with creating the dialog you want to stub `dialog_name_from_automate` to return your dialog name.

```ruby
let(:aws_dialog) do
  dialog = FactoryGirl.create(:miq_dialog_aws_provision)
  allow_any_instance_of(MiqRequestWorkflow).to receive(:dialog_name_from_automate).and_return(dialog.name)
end
```